src/managers/SessionLockManager.cpp

```diff
@@ -156,6 +156,16 @@ bool CSessionLockManager::isSurfaceSessionLock(wlr_surface* pSurface) {
 
 void CSessionLockManager::removeSessionLockSurface(SSessionLockSurface* pSLS) {
     std::erase_if(m_sSessionLock.vSessionLockSurfaces, [&](const auto& other) { return pSLS == other.get(); });
+
+    if (g_pCompositor->m_pLastFocus != nullptr)
```

Comment:
redundant != nullptr

Reply:
it will be setted to nullptr [here](https://github.com/hyprwm/Hyprland/blob/c62a50e9746009d15bc0ede0ae2b57917e789aeb/src/managers/SessionLockManager.cpp#L41) when the surface to be destroyed is focused. If i remove this condition, the focus will be changed when the surface to be destroyed is also not focused

Reply:
I meant only the `!= nullptr` part

`if (g_pCompositor->m_pLastFocus)`